src/Native/System.Native/pal_networking.h

```diff
@@ -117,6 +117,23 @@ struct LingerOption
     int32_t Seconds; // Number of seconds to linger for
 };
 
+struct IOVector
+{
+    uint8_t* Base;
+    uintptr_t Count;
+};
+
+struct MessageHeader
+{
+    uint8_t* SocketAddress;
```

Comment:
AFAICT, that assert cannot fail on any standard-compliant system. sizeof(char) must be 1. uint8_t is exactly 8 bits but is only defined on systems that can support it. If CHAR_BIT were more than 8, what would sizeof(uint8_t) be? I think such a system would not be able to provide uint8_t so nothing using it would compile.

http://en.cppreference.com/w/cpp/language/sizeof
http://en.cppreference.com/w/cpp/types/integer

I would prefer const char* for strings and uint8_t* for buffers. I think the string casts will just clutter the code.